File: middleware/python/my_middleware.py
from tyk.decorators import *
from gateway import TykGateway as tyk

@Hook
def MyPreMiddleware(request, session, spec):
    print("my_middleware: MyPreMiddleware")
    return request, session

@Hook
def AnotherPreMiddleware(request, session, spec):
    print("my_middleware: AnotherPreMiddleware")
    return request, session

@Hook
def MyPostMiddleware(request, session, spec):
    print("my_middleware: MyPostMiddleware")
    return request, session

File: middleware/lua/my_middleware.lua
  function MyPreMiddleware(request, session, spec)
  print("MyPreMiddleware, request=", request, "session=", session, "spec=", spec)
  tyk.req.set_header("myluaheader", "myluavalue")
  local headers = tyk.req.get_headers()
  print(headers)
  tyk.req.clear_header("User-Agent")
  -- print("User-Agent header:", tyk.header["User-Agent"])
  return request, session
end
